### PR TITLE
Iss4

### LINKED ISSUE #4: Setup Automated Testing
- [x] Add Automated testing using [Github Actions](https://docs.github.com/en/free-pro-team@latest/actions/quickstart) (use only one version of Node.js, for example 14.X for lesser time and resources usage);
- [x] Configure ESLint;
- [x] Add [Git Hooks](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks):
  - [x] Add pre-commit hook that checks codestyle using ESLint;
  - [x] Add pre-push hook that runs tests;
  - [x] Add automated Github Hooks configuration via npm scripts.

### PR DESCRIPTION
**Pull request by Hud-Volodymyr**
Closes #4

### CODE REVIEW

File: .github/hooks/pre-commit [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  git show :$file | eslint $file
+  if [[ $? != 0 ]] ; then
+    failed=1
+  fi
+done;
+
+if [[ $failed != 0 ]] ; then
+  echo "🚫🚫🚫 ESLint failed, git commit denied!"
+  exit $failed
+fi
```
Comment (ALEGATOR1209) [inline .github/hooks/pre-commit]: no newline at the end of the file (although, we can ignore it i suppose)

File: package.json [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
-  "version": "0.0.1",
+  "version": "0.0.2",
   "description": "Back-end for Lemon🍋",
   "main": "src/main.js",
   "scripts": {
     "test": "node test/server/serverTest.js",
     "start": "node src/main.js",
-    "testServer": "node src/testServer.js"
+    "testServer": "node src/testServer.js",
+    "hookSetup": "./set-up.sh"
```
Comment (ALEGATOR1209) [inline package.json]: Maybe, do it on npm install?

**Review by ALEGATOR1209 — DISMISSED** [review]

**Review by ALEGATOR1209 — APPROVED** [review]

**Review by MaryLynJuana — APPROVED** [review]

File: .github/hooks/pre-commit [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+if [[ "$STAGED_FILES" = "" ]]; then
+  exit 0
+fi
+
+failed=0
+for file in ${STAGED_FILES}; do
+  git show :$file | eslint $file
+  if [[ $? != 0 ]] ; then
+    failed=1
```
Comment (PaIIadium) [inline .github/hooks/pre-commit:15]: break the loop for efficiency

**Review by PaIIadium — CHANGES_REQUESTED** [review]
Add break directive in pre-commit

File: .github/hooks/pre-commit (hunk 3 above)
Comment (ALEGATOR1209) [reply .github/hooks/pre-commit:15]: Why? Let it check all files and show all errors, instead of only first one.

Comment (PaIIadium) [reply .github/hooks/pre-commit:15]: Because this script does not show errors, only "🚫🚫🚫 ESLint failed, git commit denied!"

Comment (Hud-Volodymyr) [reply .github/hooks/pre-commit:15]: As far as I understand it, it doesn't block Eslint output, meaning before my custom error messages there should be lint output for each error

Comment (PaIIadium) [reply .github/hooks/pre-commit:15]: Then, that is ok (prinyato)

**Review by PaIIadium — APPROVED** [review]

**Pull request merged**